### PR TITLE
Revise CSRF section

### PR DESCRIPTION
**Pull request by kingthorin**
This PR fixes OWASP/wstg#271.

- [x] This PR handles the issue and requires no additional PRs.
- [x] You have validated the need for this change.

**What did this PR accomplish?**

- Move Tools and References to end of document (per template).
- Add test objectives content.
- Reduce remediation content.
- Various minor edits (grammar, consistency, follow style guide, etc).
- Add some references to multi-post CSRF.
- Refresh the 3 images used in the section (fix the border on one, get up-to-date Firefox examples for the other two).

Signed-off-by: kingthorin <kingthorin@users.noreply.github.com>

### CODE REVIEW

**Comment by kingthorin** [comment]
The [link check failures](https://github.com/OWASP/wstg/pull/333/checks?check_run_id=456797408) are unrelated to the modified content.

**Comment by victoriadrake** [comment]
And now it works... apparently trying to submit edits in review mode gives a 404.

**Comment by kingthorin** [comment]
Thanks @victoriadrake

**Comment by ThunderSon** [comment]
@victoriadrake you should be able to directly update now 😄

File: document/4_Web_Application_Security_Testing/4.7_Session_Management_Testing/4.7.5_Testing_for_CSRF_WSTG-SESS-005.md [hunk 1]
```diff
@@ -2,141 +2,119 @@
...
 
 Points 1, 2, and 3 are essential for the vulnerability to be present, while point 4 facilitates the actual exploitation, but is not strictly required.
 
-Point 1) Browsers automatically send information which is used to identify a user session. Suppose *site* is a site hosting a web application, and the user *victim* has just authenticated himself to *site*. In response, *site* sends *victim* a cookie which identifies requests sent by *victim* as belonging to *victim’s* authenticated session. Basically, once the browser receives the cookie set by *site*, it will automatically send it along with any further requests directed to *site*.
+Point 1) Browsers automatically send information used to identify a user session. Suppose *site* is a site hosting a web application, and the user *victim* has just authenticated to *site*. In response, *site* sends *victim* a cookie that identifies requests sent by *victim* as belonging to *victim’s* authenticated session. Once the browser receives the cookie set by *site*, it will automatically send it along with any further requests directed to *site*.
```
Comment (ThunderSon) [inline document/4_Web_Application_Security_Testing/4.7_Session_Management_Testing/4.7.5_Testing_for_CSRF_WSTG-SESS-005.md]: What about replacing point 1) with a numbered list, and in the sentence above we update it as well?
How about as well creating it as a checklist?
- [ ] Browsers ....

What do you think?

Comment (kingthorin) [reply document/4_Web_Application_Security_Testing/4.7_Session_Management_Testing/4.7.5_Testing_for_CSRF_WSTG-SESS-005.md]: Using a checklist might be awkward for web/pdf publishing?

But going to a numbered list seems fine.

**Comment by kingthorin** [comment]
Rebased current (to pickup merged link checked)

File: document/4_Web_Application_Security_Testing/4.7_Session_Management_Testing/4.7.5_Testing_for_CSRF_WSTG-SESS-005.md [hunk 2]
```diff
@@ -2,141 +2,117 @@
...
+- HTML tags on the page resulting in automatic HTTP request execution (`img` being one of those).
+- The browser having no way to tell that the resource referenced by `img` is not a legitimate image.
+- Image loading that happens regardless of the location of the alleged image source, i.e., the form and the image itself need not be located on the same host or even the same domain. While this is a very handy feature, it makes difficult to compartmentalize applications.
 
-The problem might be even worse, since in integrated mail/browser environments simply displaying an email message containing the image would result in the execution of the request to the web application with the associated browser cookie.
+The fact that HTML content unrelated to the web application may refer to components in the application, and the fact that the browser automatically composes a valid request towards the application, allows this kind of attack. There is no way to prohibit this behavior unless it is made impossible for the attacker to specify valid application URLs. This means that valid URLs must contain information related to the user session in order to make their determination more difficult.
```
[the remark this answers is not in the document]
Comment (kingthorin) [reply document/4_Web_Application_Security_Testing/4.7_Session_Management_Testing/4.7.5_Testing_for_CSRF_WSTG-SESS-005.md]: Shrug, agreed, not my text. Will rewrite it.

File: document/4_Web_Application_Security_Testing/4.7_Session_Management_Testing/4.7.5_Testing_for_CSRF_WSTG-SESS-005.md [hunk 3]
```diff
@@ -2,141 +2,117 @@
...
+The problem here is a consequence of:
 
-It is the fact that HTML content unrelated to the web application may refer components in the application, and the fact that the browser automatically composes a valid request towards the application, that allows such kind of attacks. As no standards are defined right now, there is no way to prohibit this behavior unless it is made impossible for the attacker to specify valid application URLs. This means that valid URLs must contain information related to the user session, which is supposedly not known to the attacker and therefore make the identification of such URLs impossible.
+- HTML tags on the page resulting in automatic HTTP request execution (`img` being one of those).
+- The browser having no way to tell that the resource referenced by `img` is not a legitimate image.
+- Image loading that happens regardless of the location of the alleged image source, i.e., the form and the image itself need not be located on the same host or even the same domain. While this is a very handy feature, it makes difficult to compartmentalize applications.
```
[the remark this answers is not in the document]
Comment (kingthorin) [reply document/4_Web_Application_Security_Testing/4.7_Session_Management_Testing/4.7.5_Testing_for_CSRF_WSTG-SESS-005.md]: It’s not wrong but it is a fact that’s in general acceptance in 2020 and shouldn’t need to be said.

I’ll just drop that sentence, it doesn’t add value.

**Comment by kingthorin** [comment]
Done x2

**Pull request merged**